app/qmlV2/component/MMPhoto.qml

```diff
@@ -0,0 +1,88 @@
+import QtQuick
+import QtQuick.Controls
+import Qt5Compat.GraphicalEffects
+import "."
+import ".."
+
+Image {
+  id: control
+
+  property int hiddenPhotoCount: 0
+
+  signal clicked( var path )
+  signal showWholeGallery()
+
+  width: visible ? 120 * __dp : 0
+  height: width
+  source: visible ? modelData : ""
+  asynchronous: true
+  layer.enabled: true
+  layer {
+    effect: OpacityMask {
+      maskSource: Item {
+        width: control.width
+        height: control.height
+        Rectangle {
+          anchors.centerIn: parent
+          width: parent.width
+          height: parent.height
+          radius: 20 * __dp
+        }
+      }
+    }
+  }
+  MouseArea {
+    anchors.fill: parent
+    onClicked: control.hiddenPhotoCount > 0 ? control.showWholeGallery() : control.clicked(model.modelData)
```

Comment:
This condition does not seem to do what you want, does it? It will likely open gallery no matter which icon you click